nautobot/docs/plugins/templates.md

```diff
@@ -0,0 +1,106 @@
+# Templates for Plugins
```

Comment:
This is useful information for new core contributors as well as plugin authors. What do you think about possibly moving this under `docs/development` instead and adding a reference from `plugins/development.md` to it?

Reply:
Makes sense!